app/domain/rotation/master_rotator.rb

```diff
@@ -28,7 +31,17 @@ def rotate_every(seconds)
     end
 
     def rotate_all
-      scheduled_rotations.each(&:run)
+      # Attempt an advisory lock to prevent other servers from trying to rotate
+      # at the same time or immediately after another server rotated the secret.
+      # The same `id` needs to be used across all servers talking to the same
+      # database, but can be any unused integer.
+      id = 8675309
+      @db.try_advisory_lock(id) do
+        scheduled_rotations.each(&:run)
+      end
+    rescue => e
+      set_retry_expiration
+      log_error(e)
```

Comment:
This loop seems a tad bit odd - we still want to propagate an error up if the rotation itself fails

Reply:
Also, this really should be a warning if it was just the locking failure, not an error probably

Reply:
After reviewing the history, this was a bad copy/paste from deeper in the stack where the error handling was needed, it's been pulled out.